src/App.tsx

```diff
@@ -1,11 +1,30 @@
-import React from 'react';
-import './App.css';
+import React from "react";
+import { Route, Routes } from "react-router-dom";
+import "./App.css";
+import { UserProvider } from "./context/UserContext";
+import { Home } from "./pages/Home/Home";
+import { Landing } from "./pages/Landing/Landing";
+import { Login } from "./pages/Login/Login";
+import { NotFound } from "./pages/NotFound/NotFound";
+import { Signup } from "./pages/Signup/Signup";
 
 function App() {
   return (
-    <div className="App">
-        Hey there
-    </div>
+    <main className="App">
+      <UserProvider>
+        <Routes>
+          {/* Public routes */}
+          <Route path="/" element={<Landing />} />
+          <Route path="/login" element={<Login />} />
+          <Route path="/signup" element={<Signup />} />
+
+          {/* Private routes */}
+          <Route path="/home" element={<Home />} />
+
+          <Route path="*" element={<NotFound />} />
+        </Routes>
```

Comment:
Add route provider, and replace this code with something like this:

```tsx
        <ErrorBoundary>
            <Providers>
                <RouterProvider router={router}/>
            </Providers>
        </ErrorBoundary>
```


and the route object will be like this:

```ts
export const router = createBrowserRouter([
  {
    path: '/',
    element: <Homepage />,
    errorElement: <ErrorPage />,
  },
  {
    path: '/',
    element: <PublicRoute />,
    children: AuthRoutes,
    errorElement: <ErrorPage />,
  },
  {
    path: '/pokemons',
    element: <ProtectedRoute />,
    children: PokemonsRoutes,
    errorElement: <ErrorPage />,
  },
  {
    path: '*',
    element: <ErrorPage />,
  },
]
  , { basename: '/' }
);
```